File: WEEK 2/readability.c
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#include<ctype.h>
#include<math.h>

int main(void)
{
    int letter=0, word=1, sentence=0, grade;
    string paragraph;
    float i, L, S;
    paragraph = get_string("Text : ");

    for(int track=0, n=strlen(paragraph) ; track < n ; track++)
    {
        if (isalpha(paragraph[track]))
            letter++;   //count letters
        if (paragraph[track] == '.' || paragraph[track] == '?' || paragraph[track] == '!' || paragraph[track] == ':')
            sentence++;   //count sentences
        if (paragraph[track] == ' ')
        {
            if (isalpha(paragraph[track+1]))
                word++;  //count words
        }
    }

    i = 0.0588 * ( 100 * (float)letter / (float)word ) - 0.296 * ( 100 *(float)sentence / (float)word ) - 15.8; //calculate grade
    grade = round(i);

    if (grade > 0 && grade <= 16)
        printf("Grade %d\n",grade);
    else if(grade > 16){
        printf("Grade 16+\n");
    else
        printf("Before Grade 1\n");
}

File: WEEK6/Readability.py
def main():
    word = 1
    sentences = 0
    letter = 0
    paragraph = input("Text: ")
    print()

    for i in paragraph:
        if i.isalpha():
            letter += 1
        if i == '.' or i == '?' or i == '!':
            sentences += 1
        if i == ' ':
            word += 1

    i = 0.0588 * (100 * letter / word) - 0.296 * (100 * sentences / word) - 15.8
    print(f"{i} {letter} {word} {sentences}")
    if i >= 1 and i <= 16:
        print(f"Grade: {round(i)}")
    elif i > 16:
        print("Grade: 16+")
    else:
        print("Before Grade 1")


main()
